pkg/frontend/querymiddleware/split_and_cache.go

```diff
@@ -381,30 +387,36 @@ func (s *splitAndCacheMiddleware) fetchCacheExtents(ctx context.Context, now tim
 		extents[keyIdx] = make([]Extent, 0, len(resp.Extents))
 
 		// Filter out extents that are outside TTL.
-		for ix := range resp.Extents {
+		for ix, cachedExtent := range resp.Extents {
 			// If we don't know the query timestamp, we use the cached result.
 			// This is temporary ... after max 7 days (previous hardcoded TTL) all cached results will have query timestamp recorded.
 			usedTTL := getTTLForExtent(now, ttl, ttlForExtentsInOOOWindow, oooWindow, &resp.Extents[ix])
```

Comment:
[nit] I think we can do this here, and then `ix` is unused

```suggestion
			usedTTL := getTTLForExtent(now, ttl, ttlForExtentsInOOOWindow, oooWindow, &cachedExtent)
```

Reply:
the address that we take by `&cachedExtent` is the address of the loop variable. So all of the `getTTLForExtent` get called with the same value. I changed it to just take a value instead of a pointer, to avoid this

Reply:
Even better 🙂